### DISCUSSION TITLE
Import fastpathology annotations to qupath

### DISCUSSION THREAD

**Discussion by DREATK** [AICAN-Research/FAST-Pathology#88] [General]
Is it basicly possible, to use fastpathology for the standardized annotation of e.g. 200 melanomas and transer this data to QuPath in order to run positive cell detection?

**Comment by andreped** [comment]
Hello, @DREATK!

FastPathology does not contain any annotation functionality. QuPath is much better suited for annotating whole slide images (see [here](https://github.com/qupath/qupath)). As you are already doing downstream analysis in QuPath, it will be easier to just do everything in QuPath.

If you were to train some deep neural network model based on the annotations made in QuPath, you could import and use the generated model in FastPathology.

- More information about FastPathology is found [here](https://ieeexplore.ieee.org/document/9399433).

- To train your own deep segmentation models, you can take a look at [this paper](https://www.frontiersin.org/articles/10.3389/fmed.2021.816281/full) (includes information on how to use FastPathology and QuPath together).

---
EDIT: Or perhaps I misunderstood what you meant by "standardized annotation". Do you have a pretrained segmentation model that annotates the structure of interest? Maybe you were thinking of using the nuclei segmentation model that is available in FastPathology? But I believe the nuclei segmentation algorithm in QuPath is more robust than our simple U-Net for nuclei segmentation, that was only really trained for demonstration purposes.

If yes, then sure, that is possible. You can use FastPathology to generate annotations on the 200 WSIs of melanomas. Those predictions will be stored on disk and can be imported into QuPath as annotations. [Here](https://github.com/andreped/NoCodeSeg#getting-started) is a link to a video on how to do that.
